dmcontent/content_loader.py

```diff
@@ -35,9 +35,33 @@ def __init__(self, sections):
                 question_index += 1
                 question.number = question_index
 
+        for section in self.sections:
+            for question in section.questions:
+                if question.get("hint"):
+                    question._data["hint"] = self.question_references(question["hint"])
```

Comment:
Accessing a field on question like this feels a bit weird. I definitely think this is the right kind of approach, but potentially calling this out into a separate method might make what's going on here more self-documenting.

So for example (and I might have got the name a bit wrong) here we could do:

`question.updateHintReferences(self)`

and then all of the logic here and in your `question_references` function might be able to go onto the question class?